src/libstd/primitive_docs.rs

```diff
@@ -140,9 +142,59 @@ mod prim_bool { }
 /// [`Ok`] variant. This illustrates another behaviour of `!` - it can be used to "delete" certain
 /// enum variants from generic types like `Result`.
 ///
+/// ## Infinite loops
+///
+/// While [`Result<T, !>`] is very useful for removing errors, `!` can also be used to removed
+/// successes as well. If we think of [`Result<T, !>`] as "if this function returns, it has not
+/// errored," we get a very intuitive idea of [`Result<!, E>`] as well: if the function returns, it
+/// *has* errored.
+///
+/// For example, consider the case of a simple web server, which can be simplified to:
+///
+/// ```ignore (hypothetical-example)
+/// loop {
+///     let (client, request) = get_request().expect("disconnected");
+///     let response = request.process();
+///     response.send(client);
+/// }
+/// ```
+///
+/// Currently, this isn't ideal, because we simply panic whenever we fail to get a new connection.
+/// Instead, we'd like to keep track of this error, like this:
+///
+/// ```ignore (hypothetical-example)
+/// loop {
+///     match get_request() {
+///         Err(err) => break err,
+///         Ok((client, request)) => {
+///             let response = request.process();
+///             response.send(client);
+///         },
+///     }
+/// }
+/// ```
+///
+/// Now, when the server disconnects, we exit the loop with an error instead of panicking. While it
+/// might be intuitive to simply return the error, we might want to wrap it in a [`Result<!, E>`] 
+/// instead:
+///
+/// ```ignore (hypothetical-example)
+/// fn server_loop() -> Result<!, ConnectionError> {
```

Comment:
I really like this example :+1:  A thought: with https://github.com/rust-lang/rust/pull/49162, it could even be 
```rust
fn main() -> Result<!, ConnectionError>
```

Also, I think the `Ok(` isn't needed because loop returns `!`, which will coerce to the `Result`: https://play.rust-lang.org/?gist=1cc0b7e8032bec565768ab95f42c454d&version=nightly

Reply:
My main reasoning for not making this `main` is that the idea is that you might in principle want a larger loop that handles the error. Also, while I agree that the loop can be coerced to `Ok`, I think that wrapping in an `Ok` is best to draw parallels to a non-infinite loop. If it seems more clear the other way though, I'll definitely change it.

Reply:
I'd prefer to leave off the Ok, but I can agree that this being not main makes a lot of sense

Reply:
I'll leave off the `Ok` and explain the coercion.